src/metrics/metrics.go

```diff
@@ -160,72 +175,16 @@ func populateCustomMetrics(instanceEntity *integration.Entity, connection *conne
 	var rowCount = 0
 	for rows.Next() {
 		rowCount++
-		row := make(map[string]interface{})
-		err := rows.MapScan(row)
-		if err != nil {
+		values := make([]string, len(columns))                 // All values are represented as strings (the corresponding conversion is handled while scanning)
+		valuesForScanning := make([]interface{}, len(columns)) // the `rows.Scan` function requires an array of interface{}
+		for i := range valuesForScanning {
+			valuesForScanning[i] = &values[i]
+		}
+		if err := rows.Scan(valuesForScanning...); err != nil {
```

Comment:
One possible flow is that before we were not panicking if it as not possible to parse it to string:
```
name, ok = nameInterface.(string)
	if !ok {				
              log.Error("Non-string type %T for custom query 'metric_name' column", nameInterface)
```

Now I am not sure how `rows.Scan` would behave. I wonder if it is not safer to read everything in []interface{} and do the cast manually

Reply:
However, I think it is safe according to the docs

Reply:
I had doubts as well, but I checked the code behind `Scan` and I realized that the conversion is not as simple as `value.(string)`, it performs particular conversions depending on the original and target types. 

I also considered keeping the 'read everything to interface{}` behavior, but then DECIMAL values where got as `[]uint8` and we needed additional information concerning the column type to convert it (and that information is not always available).

I think reading everything as string is safe and the implementation is also simpler.